tests/unittests/sources/test_azure.py

```diff
@@ -2240,38 +2240,71 @@ def partitions_on_device(devpath):
             # return sorted by partition number
             return sorted(ret, key=lambda d: d[0])
 
-        def mount_cb(
-            device, callback, mtype, update_env_for_mount, log_error=False
-        ):
-            self.assertEqual("ntfs", mtype)
-            self.assertEqual("C", update_env_for_mount.get("LANG"))
-            p = self.tmp_dir()
-            for f in bypath.get(device).get("files", []):
-                write_file(os.path.join(p, f), content=f)
-            return callback(p)
-
         def has_ntfs_fs(device):
             return bypath.get(device, {}).get("fs") == "ntfs"
 
         p = MOCKPATH
         self._domock(p + "_partitions_on_device", "m_partitions_on_device")
         self._domock(p + "_has_ntfs_filesystem", "m_has_ntfs_filesystem")
-        self._domock(p + "util.mount_cb", "m_mount_cb")
         self._domock(p + "os.path.realpath", "m_realpath")
         self._domock(p + "os.path.exists", "m_exists")
         self._domock(p + "util.SeLinuxGuard", "m_selguard")
 
         self.m_exists.side_effect = lambda p: p in bypath
         self.m_realpath.side_effect = realpath
         self.m_has_ntfs_filesystem.side_effect = has_ntfs_fs
-        self.m_mount_cb.side_effect = mount_cb
         self.m_partitions_on_device.side_effect = partitions_on_device
         self.m_selguard.__enter__ = mock.Mock(return_value=False)
         self.m_selguard.__exit__ = mock.Mock()
 
+        return bypath
+
+    M_PATH = "cloudinit.util."
+
+    @mock.patch(M_PATH + "subp.subp")
+    def test_ntfs_mount_logs(self, m_subp):
+        """can_dev_be_reformatted does not log errors in case of
+        unknown filesystem 'ntfs'."""
+        self.patchup(
+            {
+                "/dev/sda": {
+                    "partitions": {
+                        "/dev/fake0": {"num": 1, "fs": "ntfs", "files": []}
+                    }
+                }
+            }
+        )
+
+        log_msg = (
+            "Failed to mount device: '/dev/fake0' with type: "
+            "'ntfs' using mount command:"
+        )
+
+        m_subp.side_effect = subp.ProcessExecutionError(
+            "", "unknown filesystem type 'ntfs'"
+        )
+
+        dsaz.can_dev_be_reformatted("/dev/sda", preserve_ntfs=False)
+        self.assertNotIn(log_msg, self.logs.getvalue())
+
+    def mount(self, bypath):
```

Comment:
suggest renaming to _domock_mount_cb() or something to that effect so it doesn't read like we're mounting